core/src/main/java/org/elasticsearch/index/query/QueryStringQueryParser.java

```diff
@@ -100,148 +100,109 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
         }
-        qpSettings.defaultAnalyzer(context.mapperService().searchAnalyzer());
-        qpSettings.defaultQuoteAnalyzer(context.mapperService().searchQuoteAnalyzer());
-
-        if (qpSettings.escape()) {
-            qpSettings.queryString(org.apache.lucene.queryparser.classic.QueryParser.escape(qpSettings.queryString()));
-        }
-
-        MapperQueryParser queryParser = context.queryParser(qpSettings);
 
-        try {
-            Query query = queryParser.parse(qpSettings.queryString());
-            if (query == null) {
-                return null;
-            }
-            if (qpSettings.boost() != QueryParserSettings.DEFAULT_BOOST) {
-                query.setBoost(query.getBoost() * qpSettings.boost());
-            }
-            query = fixNegativeQueryIfNeeded(query);
-            if (query instanceof BooleanQuery) {
-                query = Queries.applyMinimumShouldMatch((BooleanQuery) query, qpSettings.minimumShouldMatch());
-            }
-            if (queryName != null) {
-                context.addNamedQuery(queryName, query);
-            }
-            return query;
-        } catch (org.apache.lucene.queryparser.classic.ParseException e) {
-            throw new QueryParsingException(parseContext, "Failed to parse query [" + qpSettings.queryString() + "]", e);
-        }
+        QueryStringQueryBuilder queryStringQuery = new QueryStringQueryBuilder(queryString);
+        queryStringQuery.fields(fieldsAndWeights);
+        queryStringQuery.defaultField(defaultField);
+        queryStringQuery.defaultOperator(defaultOperator);
+        queryStringQuery.analyzer(analyzer);
+        queryStringQuery.quoteAnalyzer(quoteAnalyzer);
+        queryStringQuery.allowLeadingWildcard(allowLeadingWildcard);
+        queryStringQuery.autoGeneratePhraseQueries(autoGeneratePhraseQueries);
+        queryStringQuery.maxDeterminizedStates(maxDeterminizedStates);
+        queryStringQuery.lowercaseExpandedTerms(lowercaseExpandedTerms);
+        queryStringQuery.enablePositionIncrements(enablePositionIncrements);
+        queryStringQuery.escape(escape);
+        queryStringQuery.useDisMax(useDisMax);
+        queryStringQuery.fuzzyPrefixLength(fuzzyPrefixLength);
+        queryStringQuery.fuzzyMaxExpansions(fuzzyMaxExpansions);
+        queryStringQuery.fuzzyRewrite(fuzzyRewrite);
+        queryStringQuery.phraseSlop(phraseSlop);
+        queryStringQuery.fuzziness(fuzziness);
+        queryStringQuery.tieBreaker(tieBreaker);
+        queryStringQuery.analyzeWildcard(analyzeWildcard);
+        queryStringQuery.rewrite(rewrite);
+        queryStringQuery.minimumShouldMatch(minimumShouldMatch);
+        queryStringQuery.quoteFieldSuffix(quoteFieldSuffix);
+        queryStringQuery.lenient(lenient);
+        queryStringQuery.timeZone(timeZone);
+        queryStringQuery.locale(locale);
+        queryStringQuery.boost(boost);
+        queryStringQuery.queryName(queryName);
+        return queryStringQuery;
```

Comment:
I think that we went for not having conditional assignments here, but simply set. Not sure how much this would buy us. It would be better to be able to set things to the query directly actually and avoid having intermediate variables, but some fields are final and we might find them only at the end of parsing, so that's a no-go. I think this is as good as it gets?